app/Http/Controllers/AdminDash.php

```diff
@@ -796,6 +801,23 @@ public function storeFile(Request $request) {
             'file' => 'required'
         ]);
 
+        $existing_permalink = $request->input('existingPermalinks');
+        if ($existing_permalink != null) {
+            $existing_file = \App\File::select('id', 'path')->where('permalink', $existing_permalink)->first();
+            if ($existing_file && $existing_file->id) {
+                $existing_file_path = $existing_file->path;
+                $storage_path = ltrim($existing_file_path, '/');
+        
+                if (Storage::exists($storage_path)) {
```

Comment:
I think it's a good idea to throw a check in here to ensure that we're deleting a file and not a directory. May I suggest:
`if (Storage::exists($storage_path) && !is_dir($storage_path)) {`
I don't know when this would come into play, but it is a nice check to ensure that we're only doing delete operations on a file, as intended.